OONIProbeUnitTests/APITest.m

```diff
@@ -102,6 +102,35 @@ - (void)testDeleteJsons {
     }];
 }
 
+- (void)testJsonFromExplorer {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"testJsonFromExplorer"];
+    [TestUtility downloadJson:JSON_URL
+                    onSuccess:^(NSDictionary *urls) {
+                        XCTAssert(true);
+                        [expectation fulfill];
+                    } onError:^(NSError *error) {
+                        XCTAssert(false);
+                        [expectation fulfill];
+                    }];
+    [self waitForExpectationsWithTimeout:10.0 handler:^(NSError *err) {
+      XCTAssert(err == nil);
+    }];
+}
+
+- (void)testMalformedURL {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"testMalformedURL"];
+    [TestUtility downloadJson:@"http://www.google.it"
```

Comment:
I meant something like `http://\t`. This should break the URL parser. (I do not understand why the URL you provided is malformed, so maybe I'm missing something, tho &mdash; can you explain?)

Reply:
This test with the wrong URL is probably still good, but it seems to me to be another test. The URL seems to be well formed (it's a valid URL) however it's not the right URL for fetching measurements.

Reply:
Anything that is not a json makes the function going to onError

Reply:
I'm concerned about a much earlier error, where URL parsing fails. A way to do that is to feed the URL parser with an invalid URL like `http://\t`.